File: frontend-react/src/features/planoContas/planoContasSystemGroups.js
function normalizeText(value) {
  return String(value ?? '')
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '')
    .replace(/\s*-\s*/g, ' - ')
    .replace(/\s+/g, ' ')
    .trim()
    .toUpperCase();
}

export const SYSTEM_PROTECTED_GROUP_NAMES = Object.freeze([
  'CUSTO FIXO PESSOAL',
  'CUSTO FIXO PROFISSIONAL',
  'CUSTO VARIAVEL PESSOAL',
  'CUSTO VARIAVEL PROFISSIONAL',
  'INVESTIMENTO - EMPRESA',
  'INVESTIMENTO - PESSOAL',
]);

export const SYSTEM_PROTECTED_GROUP_ALIASES = Object.freeze([
  'CUSTO VARIVAVEL PROFISSIONAL',
  'INVESTIMENTOS - EMPRESA',
  'INVESTIMENTOS - PESSOAL',
]);

export function normalizeFinancialGroupName(value) {
  return normalizeText(value);
}

export function normalizePlanoContasSystemGroupName(name) {
  return normalizeText(name);
}

export function isSystemProtectedGroupName(value) {
  const normalized = normalizeFinancialGroupName(value);
  return SYSTEM_PROTECTED_GROUP_NAMES.includes(normalized) || SYSTEM_PROTECTED_GROUP_ALIASES.includes(normalized);
}

export function isSystemProtectedGroup(group) {
  return isSystemProtectedGroupName(group?.nome);
}

export function isPlanoContasSystemProtectedGroup(groupOrName) {
  if (typeof groupOrName === 'string') {
    return isSystemProtectedGroupName(groupOrName);
  }
  return isSystemProtectedGroup(groupOrName);
}

File: backend/services/plano_contas_system_groups.py
import re
import unicodedata

SYSTEM_PROTECTED_GROUP_NAMES = frozenset(
    {
        "CUSTO FIXO PESSOAL",
        "CUSTO FIXO PROFISSIONAL",
        "CUSTO VARIAVEL PESSOAL",
        "CUSTO VARIAVEL PROFISSIONAL",
        "INVESTIMENTO - EMPRESA",
        "INVESTIMENTO - PESSOAL",
    }
)

SYSTEM_PROTECTED_GROUP_ALIASES = frozenset(
    {
        "CUSTO VARIVAVEL PROFISSIONAL",
        "INVESTIMENTOS - EMPRESA",
        "INVESTIMENTOS - PESSOAL",
    }
)


def normalize_financial_group_name(value: str | None) -> str:
    text = str(value or "").strip()
    if not text:
        return ""
    text = unicodedata.normalize("NFD", text)
    text = "".join(ch for ch in text if unicodedata.category(ch) != "Mn")
    text = re.sub(r"\s*-\s*", " - ", text)
    text = re.sub(r"\s+", " ", text)
    return text.strip().upper()


def is_system_protected_group_name(value: str | None) -> bool:
    normalized = normalize_financial_group_name(value)
    return bool(normalized) and (
        normalized in SYSTEM_PROTECTED_GROUP_NAMES or normalized in SYSTEM_PROTECTED_GROUP_ALIASES
    )
